### PR TITLE
fix: 🐛 preserve original stack trace

### LINKED ISSUE #2705: [Bug]: cli doesn't show config loading error details
### Reproduction link or steps

https://stackblitz.com/edit/github-xv4yss-5lbkmh?file=rolldown.config.mjs

Since https://github.com/rolldown/rolldown/pull/2688/, CLI doesn't surface the original error which happened during config loading.

### What is expected?

Like v0.14.0

image

### What is actually happening?

image

### System Info

```Shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    rolldown: nightly => 0.14.0-snapshot-d18e8cb-20241112003342
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

File: packages/rolldown/src/cli/utils.ts [hunk 1]
```diff
@@ -45,12 +45,17 @@ export async function ensureConfig(configPath: string): Promise<ConfigExport> {
...
     configExports = await import(fileUrl)
-  } catch (err) {
+  } catch (err: any) {
     let errorMessage = 'Error happened while loading config.'
     if (!isSupportedFormat(configPath)) {
       errorMessage += ` Unsupported config format. Expected: \`${SUPPORTED_CONFIG_FORMATS.join(',')}\` but got \`${nodePath.extname(configPath)}\``
     }
-    throw new Error(errorMessage, { cause: err })
+    let newError = new Error(errorMessage)
+    ;(newError as any).original_error = err
```
Comment (hyf0) [inline packages/rolldown/src/cli/utils.ts:54]: We could still use `cause` field, since it's a standard.

**Comment by hyf0** [comment]
I test locally a little. It seems that `console.log(err)` could print the `cause` field. But if the error is cautght by the nodejs itself, it just print the error without casue.
Maybe code like that

```js
try {
  ...everthing()
} catch (err) {
  console.error(err);
  process.exit(1);
}
```

could help.

However, merging stacks is another way I saw  in https://www.npmjs.com/package/merge-error-cause.

They both solve the problem. Anyway, I leave the choices to you.

**Review by hyf0 — APPROVED** [review]

**Comment by IWANABETHATGUY** [comment]
This is intent to solve https://github.com/rolldown/rolldown/issues/2705

**Comment by hyf0** [comment]
Looks like 

https://github.com/rolldown/rolldown/blob/15ac3de64d13599f2307e992b537f4328c46c9e9/packages/rolldown/src/cli/index.ts#L31

it's the problem of `logger.error`. Nodejs will also print out `cause`.

**Comment by IWANABETHATGUY** [comment]
> I test locally a little. It seems that `console.log(err)` could print the `cause` field. But if the error is cautght by the nodejs itself, it just print the error without casue. Maybe code like that
> 
> ```js
> try {
>   ...everthing()
> } catch (err) {
>   console.error(err);
>   process.exit(1);
> }
> ```
> 
> could help.
> 
> However, merging stacks is another way I saw in https://www.npmjs.com/package/merge-error-cause.
> 
> They both solve the problem. Anyway, I leave the choices to you.

I just use they way described in https://stackoverflow.com/a/42755876/28261492,

**Comment by IWANABETHATGUY** [comment]
> Looks like
> 
> https://github.com/rolldown/rolldown/blob/15ac3de64d13599f2307e992b537f4328c46c9e9/packages/rolldown/src/cli/index.ts#L31
> 
> it's the problem of `logger.error`. Nodejs will also print out `cause`.

Yeah

**Comment by hyf0** [comment]
https://github.com/unjs/consola/issues/202 Looks like consola has no energy to solve this. Maybe switch a more predicable logger?

**Comment by IWANABETHATGUY** [comment]
> [unjs/consola#202](https://github.com/unjs/consola/issues/202) Looks like consola has no energy to solve this. Maybe switch a more predicable logger?

agree

**Pull request closed without merging**